File: version/1.8.9/src/main/java/com/github/sorusclient/client/adapter/v1_8_9/BossBarImpl.java
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter.v1_8_9;

import com.github.sorusclient.client.adapter.BossBarColor;
import com.github.sorusclient.client.adapter.IBossBar;
import v1_8_9.net.minecraft.entity.boss.BossBar;

public class BossBarImpl implements IBossBar {

    @Override
    public String getName() {
        return BossBar.name;
    }

    @Override
    public double getPercentage() {
        return BossBar.percent;
    }

    @Override
    public BossBarColor getColor() {
        return BossBarColor.PURPLE;
    }

}

File: src/main/kotlin/com/github/sorusclient/client/adapter/v1_18_1/BossBarImpl.kt
package com.github.sorusclient.client.adapter.v1_18_1

import com.github.sorusclient.client.adapter.IBossBar
import v1_18_1.net.minecraft.client.gui.hud.ClientBossBar

class BossBarImpl(private val bossBar: ClientBossBar) : IBossBar {

    override val name: String
        get() = bossBar.name.string

    override val percentage: Double
        get() = bossBar.percent.toDouble()

}